examples/notebooks/beam-ml/rag_usecase/opensearch_connector.py

```diff
@@ -0,0 +1,372 @@
+        self.batch.append(element)
+        self.batch_counter += 1
+        if self.batch_counter >= self.batch_size:
+            self._flush()
+        yield element
+
+    def _flush(self):
+        if self.batch_counter == 0:
+            return
+
+        with _InsertDocOpenSearchSink(self.host, self.port, self.username, self.password) as sink:
+            sink.write(self.batch)
+            self.batch_counter = 0
+            self.batch = list()
+
+
+class _InsertDocOpenSearchSink(object):
+    """Class where we create Opensearch client
+    and write insertion logic in Opensearch
+    """
+
+    def __init__(self,
+                 host: str,
+                 port: int,
+                 username: str,
+                 password: str
+                 ):
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.client = None
+
+    def _create_client(self):
+        if self.client is None:
+            http_auth = [self.username, self.password]
+            self.client = OpenSearch(hosts=[f'{self.host}:{self.port}'],
+                                     http_auth=http_auth,
+                                     verify_certs=False)
+
+    def write(self, elements):
+        self._create_client()
+        documents = []
+        logger.info(f'Adding Docs in DB: {len(elements)}')
+        for element in elements:
+            documents.extend([{
+                "index": {
+                    "_index": "embeddings-index",
+                    "_id": str(element["id"]),
+                }
+            }, {
+                "url": element["url"],
+                "title": element["title"],
+                "text": element["text"],
+                "section_id": element["section_id"]
+            }])
+
+        self.client.bulk(body=documents, refresh=True)
+
+    def __enter__(self):
+        self._create_client()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.client is not None:
+            self.client.close()
+
+"""This module implements IO classes to read text Embeddings in Opensearch.
+Insert Embedding in Opensearch :
+-----------------
+:class:`InsertEmbeddingInOpensearch` is a ``PTransform`` that writes key and values to a
+configured sink, and the write is conducted through a Opensearch pipeline.
+
+The ptransform works by getting the first and second elements from the input,
+this means that inputs like `[k,v]` or `(k,v)` are valid.
+
+Example usage::
+
+  pipeline | WriteToOpensearch(host='localhost',
+                          port=6379,
+                          batch_size=100)
+
+
+No backward compatibility guarantees. Everything in this module is experimental.
+"""
+
+
+class InsertEmbeddingInOpenSearch(PTransform):
+    """WriteToOpensearch is a ``PTransform`` that writes a ``PCollection`` of
+    key, value tuple or 2-element array into a Opensearch server.
```

Comment:
Could you please correct this doc string?

Reply:
Done.